### PR TITLE
Use useInterval() hook to refetch audit status in lighthouse plugin WIP

### PR DESCRIPTION
**Pull request by elchocarrero**
## Hey, I just made a Pull Request!

Fetches website using useInterval() until audit is completed or failed. Implements  #522

#### :heavy_check_mark: Checklist

- [x] All tests are passing `yarn test`
- [ ] Screenshots attached (for UI changes)
- [ ] Relevant documentation updated
- [ ] Prettier run on changed files
- [ ] Tests added for new functionality
- [ ] Regression tests added for bug fixes

### CODE REVIEW

**Comment by braulio-balanza** [comment]
WIP, need to update tests to account for the additional components

**Comment by stefanalund** [comment]
Thanks! Could you update the PR title? Also added @fastfrwrd as reviewer since he wrote the original version of this plugin.

**Comment by braulio-balanza** [comment]
There are a couple of things. 
- It would be cool to have a transition CSS effect whenever the audit is re-fetched. I was trying to use ```<Fade>``` from material UI, but the ```AuditRow``` is not re-rendering when the states are updated, only the children re-render. 
- I would also like to fix some of the ```yarn test``` warnings, specially an error relating to ```an update to AuditList inside a test was not wrapped in act(...).```
- Would also like to refactor as @fastfrwrd 's initial suggestion to a flux architecture

File: plugins/lighthouse/.eslintrc.js [hunk 1]
```diff
@@ -1,3 +1,10 @@
 module.exports = {
+  plugins: [
+  "react-hooks"
```
Comment (freben) [inline plugins/lighthouse/.eslintrc.js:3]: may want to set your editor to run prettier on save

File: plugins/lighthouse/src/components/AuditList/AuditListTable.tsx [hunk 2]
```diff
@@ -66,34 +45,13 @@ const useStyles = makeStyles(theme => ({
...
-}
-
 export const AuditListTable: FC<{ items: Website[] }> = ({ items }) => {
   const classes = useStyles();
   const categorySparklines: Record<string, SparklinesDataByCategory> = useMemo(
     () =>
       items.reduce(
         (res, item) => ({
-          ...res,
+           ...res,
```
Comment (freben) [inline plugins/lighthouse/src/components/AuditList/AuditListTable.tsx:54]: accidental change; set editor to prettier on save

File: plugins/lighthouse/src/components/Audit/index.tsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+export const Audit: FC<{
+  website: Website;
+  categorySparkline: SparklinesDataByCategory;
+}> = ({ website, categorySparkline }) => {
+  const lighthouseApi = useApi(lighthouseApiRef);
+  const fetchWebsite = async (auditId: string) => {
+    const response = await lighthouseApi.getWebsiteForAuditId(auditId);
+    return response;
+  };
+  const [delay] = useState(5000);
```
Comment (freben) [inline plugins/lighthouse/src/components/Audit/index.tsx:36]: why is this a state?

File: plugins/lighthouse/src/components/Audit/index.tsx [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  const [isIntervalRunning, toggleInterval] = useBoolean(false);
+  const [websiteState, setWebsiteState] = useState(website);
+  const [sparklineState, setSparklineState] = useState(categorySparkline);
+
+  useEffect(() => {
+    if (websiteState.lastAudit.status === 'RUNNING') {
+      toggleInterval(true);
+    } else toggleInterval(false);
+  });
```
Comment (freben) [inline plugins/lighthouse/src/components/Audit/index.tsx:45]: yeah this runs on every render and is not necessary; check above comment

File: plugins/lighthouse/src/components/Audit/index.tsx [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+  website: Website;
+  categorySparkline: SparklinesDataByCategory;
+}> = ({ website, categorySparkline }) => {
+  const lighthouseApi = useApi(lighthouseApiRef);
+  const fetchWebsite = async (auditId: string) => {
+    const response = await lighthouseApi.getWebsiteForAuditId(auditId);
+    return response;
+  };
+  const [delay] = useState(5000);
+  const [isIntervalRunning, toggleInterval] = useBoolean(false);
```
Comment (freben) [inline plugins/lighthouse/src/components/Audit/index.tsx:37]: I don't think you need this as a state at all. You could just do

```
useInterval(runRefresh, websiteState?.lastAudit?.status === 'RUNNING' ? 5000 : 0);
```

File: plugins/lighthouse/src/components/Audit/index.tsx [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+} from '../../utils';
+import { AuditRow } from './AuditRow';
+export const LIMIT = 10;
+
+export const Audit: FC<{
+  website: Website;
+  categorySparkline: SparklinesDataByCategory;
+}> = ({ website, categorySparkline }) => {
+  const lighthouseApi = useApi(lighthouseApiRef);
+  const fetchWebsite = async (auditId: string) => {
```
Comment (freben) [inline plugins/lighthouse/src/components/Audit/index.tsx:32]: This one is not necessary as a standalone thing. You may want to make a separate `const runRefresh = () => {}` that does this plus what's inside the useInterval below. Or just move it all inside useInterval. Matter of taste

File: plugins/lighthouse/.eslintrc.js (hunk 1 above)
Comment (braulio-balanza) [reply plugins/lighthouse/.eslintrc.js:3]: This was actually me. I added the eslint-react-hooks plugin. I don't know if it is ok to leave it in

File: plugins/lighthouse/src/components/AuditList/AuditListTable.tsx (hunk 2 above)
Comment (braulio-balanza) [reply plugins/lighthouse/src/components/AuditList/AuditListTable.tsx:54]: This was actually an accident lol

**Comment by braulio-balanza** [comment]
Thank you @freben for your review! Just fixed the issues you pointed out.

File: plugins/lighthouse/.eslintrc.js (hunk 1 above)
Comment (freben) [reply plugins/lighthouse/.eslintrc.js:3]: Right, that's OK, I just meant that it wasn't at the right indentation level so your editor may need some reconfiguring (or run reformat manually) :)

Comment (Rugvip) [reply plugins/lighthouse/.eslintrc.js:3]: Freben is talking about the indentation on line 3, it's not far enough.

Regarding the rule I actually thought this was a part of `spotify/web-scripts`, @fastfrwrd do you want to add react-hooks there? I find it catches a lot of errors.

Comment (fastfrwrd) [reply plugins/lighthouse/.eslintrc.js:3]: huh, I really thought it was already in there! I've opened https://github.com/spotify/web-scripts/issues/305 to track.

File: plugins/lighthouse/src/components/Audit/index.tsx [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+import React, { FC, useState } from 'react';
+import { useInterval } from 'react-use';
+import { Website, lighthouseApiRef } from '../../api';
+import { useApi } from '@backstage/core';
+import {
+  SparklinesDataByCategory,
+  buildSparklinesDataForItem,
+} from '../../utils';
+import { AuditRow } from './AuditRow';
+export const LIMIT = 10;
```
Comment (Rugvip) [inline plugins/lighthouse/src/components/Audit/index.tsx:25]: This isn't used right?

**Review by Rugvip — COMMENTED** [review]
Code LGTM, what do you think @fastfrwrd?

File: plugins/lighthouse/src/components/CreateAudit/index.tsx [hunk 8]
```diff
@@ -84,7 +84,7 @@ const CreateAudit: FC<{}> = () => {
     } finally {
       setSubmitting(false);
     }
-  }, [url, emulatedFormFactor, lighthouseApi, setSubmitting]);
+  }, [url, emulatedFormFactor, lighthouseApi, setSubmitting, errorApi, history]);
```
Comment (fastfrwrd) [inline plugins/lighthouse/src/components/CreateAudit/index.tsx:87]: thanks for fixing this!

**Review by fastfrwrd — APPROVED** [review]
Looks great to me!

**Review by stefanalund — APPROVED** [review]

**Pull request merged**